api/index.ts

```diff
@@ -0,0 +1,209 @@
+import bcd, {
+  BrowserName,
+  Browsers,
+  BrowserStatement,
+  CompatStatement,
+  Identifier,
+  SimpleSupportStatement,
+  SupportStatement,
+  VersionValue,
+} from "@mdn/browser-compat-data";
+
+interface Data {
+  data: IdentifierExtended;
+  browsers: Browsers;
+}
+
+interface SimpleSupportStatementExtended extends SimpleSupportStatement {
+  release_date?: string;
+  version_last?: VersionValue;
+}
+
+interface CompatStatementExtended extends CompatStatement {
+  support: Partial<Record<BrowserName, SimpleSupportStatementExtended[]>>;
+}
+
+interface IdentifierExtended {
+  [key: string]: IdentifierExtended | CompatStatementExtended;
+}
+
+const {
+  __meta: { version: bcdVersion },
+  browsers,
+  ...bcdAPIs
+} = bcd;
+
+export { bcdVersion, bcdAPIs };
+
+const filteredBrowsers = Object.fromEntries(
+  Object.entries(browsers).map(([name, statement]) => [
+    name,
+    {
+      ...statement,
+      releases: Object.fromEntries(
+        Object.entries(statement.releases).filter(
+          ([_, statement]) => statement.status !== "retired"
+        )
+      ),
+    },
+  ])
+) as Browsers;
+
+export default function getDataForPath(path: string): Data | void {
+  const subtree = path
+    .split(".")
+    .reduce<Identifier | undefined>((prev, curr) => prev?.[curr], bcdAPIs);
+  if (subtree) {
+    return {
+      data: walk(subtree, path),
+      browsers: filteredBrowsers,
+    };
+  }
+}
+
+export function walk(
+  tree: Identifier,
+  path: string,
+  callback?: (data: Data, path: string) => void
+): IdentifierExtended {
+  const extendedTree: IdentifierExtended = Object.fromEntries(
+    Object.entries(tree).map(([key, subtree]) => [
+      key,
+      key === "__compat"
+        ? extendCompatStatement(subtree as CompatStatement)
+        : walk(subtree as Identifier, `${path}.${key}`, callback),
+    ])
+  );
+
+  const data: Data = {
+    data: extendedTree,
```

Comment:
```suggestion
    data: extendedTree,
    query: path.slice(1),
```
and update the type (that's in the `bcd.json`. It's redundant but might come in handy.